pkg/library/lifecycle/poststart.go

```diff
@@ -99,9 +146,85 @@ func processCommandsForPostStart(commands []dw.Command) (*corev1.LifecycleHandle
 			Command: []string{
 				"/bin/sh",
 				"-c",
-				fmt.Sprintf(redirectOutputFmt, joinedCommands),
+				fmt.Sprintf(noTimeoutRedirectOutputFmt, joinedCommands),
 			},
 		},
 	}
 	return handler, nil
 }
+
+// buildUserScript takes a list of DevWorkspace commands and constructs a single
+// shell script string that executes them sequentially.
+func buildUserScript(commands []dw.Command) (string, error) {
+	var commandScriptLines []string
+	for _, command := range commands {
+		execCmd := command.Exec
+		if execCmd == nil {
+			// Should be caught by earlier validation, but good to be safe
+			return "", fmt.Errorf("exec command is nil for command ID %s", command.Id)
+		}
+		if len(execCmd.Env) > 0 {
+			return "", fmt.Errorf("env vars in postStart command %s are unsupported", command.Id)
```

Comment:
Maybe I'm missing something obvious, but why are env vars in the postStart command not supported?

Reply:
You're right, there's nothing here that prevents us from passing env variables to the postStart command. I fixed this issue.